contentcuration/contentcuration/frontend/shared/vuex/snackbar/index.js

```diff
@@ -0,0 +1,36 @@
+export default {
+  state: {
+    isVisible: false,
+    options: {
+      text: '',
+      autoDismiss: true,
+    },
+  },
+  getters: {
+    snackbarIsVisible(state) {
+      return state.isVisible;
+    },
+    snackbarOptions(state) {
+      return state.options;
+    },
+  },
+  mutations: {
+    CORE_CREATE_SNACKBAR(state, snackbarOptions = {}) {
+      // reset
+      state.isVisible = false;
+      state.options = {};
+      // set new options
+      state.isVisible = true;
+      // options include text, autoDismiss, duration, actionText, actionCallback,
+      // hideCallback, bottomPosition
+      state.options = snackbarOptions;
+    },
+    CORE_CLEAR_SNACKBAR(state) {
+      state.isVisible = false;
+      state.options = {};
+    },
+    CORE_SET_SNACKBAR_TEXT(state, text) {
+      state.options.text = text;
+    },
+  },
```

Comment:
Two notes:

1. It's sometimes a bit nicer to expose vuex actions as the primary API rather than mutations
2. Recommend making the set of available options constrained and explicit.

So in particular this might look like:

```javascript
actions: {
  showSnackbar(store, {text, autoDismiss, duration, actionText, actionCallback}) {
    store.commit('CORE_CREATE_SNACKBAR', {text, autoDismiss, duration, actionText, actionCallback});
  },
  clearSnackbar(store) {
    store.commit('CORE_CLEAR_SNACKBAR');
  },
}
```

Then the recommended use is to call these actions